src/azure/storage/blob/blob_stream_builder.rs

```diff
@@ -0,0 +1,409 @@
+use azure::core::errors::AzureError;
+use azure::core::lease::LeaseId;
+use azure::core::range::Range;
+use azure::core::{
+    BlobNameRequired, BlobNameSupport, ClientRequestIdOption, ClientRequestIdSupport, ClientRequired, ContainerNameRequired,
+    ContainerNameSupport, LeaseIdOption, LeaseIdSupport, No, RangeRequired, RangeSupport, SnapshotOption, SnapshotSupport, TimeoutOption,
+    TimeoutSupport, ToAssign, Yes,
+};
+use azure::storage::blob::requests::GetBlobBuilder;
+use azure::storage::client::Client;
+use chrono::{DateTime, Utc};
+use futures::prelude::*;
+use futures::stream;
+use std::marker::PhantomData;
+
+#[derive(Debug, Clone)]
+pub struct BlobStreamBuilder<'a, ContainerNameSet, BlobNameSet, RangeSet>
+where
+    ContainerNameSet: ToAssign,
+    BlobNameSet: ToAssign,
+    RangeSet: ToAssign,
+{
+    client: &'a Client,
+    p_container_name: PhantomData<ContainerNameSet>,
+    p_blob_name: PhantomData<BlobNameSet>,
+    p_range: PhantomData<RangeSet>,
+    container_name: Option<&'a str>,
+    blob_name: Option<&'a str>,
+    snapshot: Option<DateTime<Utc>>,
+    timeout: Option<u64>,
+    range: Option<&'a Range>,
+    lease_id: Option<&'a LeaseId>,
+    client_request_id: Option<&'a str>,
```

Comment:
Since in the end a stream is just a bunch of `get`s I think this field can be removed (I've also noticed you did not use it anyway in the `finalize` function). What do you think?

Reply:
Do you mean the lease_id and client_request ?